include/spirv-tools/instrument.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2018 The Khronos Group Inc.
+// Copyright (c) 2018 Valve Corporation
+// Copyright (c) 2018 LunarG Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDE_SPIRV_TOOLS_INSTRUMENT_HPP_
+#define INCLUDE_SPIRV_TOOLS_INSTRUMENT_HPP_
+
+namespace spvtools {
+
+// Error Codes
+// These are utilized in debug output record streams and identify which
+// validation error has occurred and written a specific record.
+static const uint32_t kInstErrorBindlessBounds = 0;
```

Comment:
Do all of these need to be in the public api? If not, this should probably live in the opt/ directory and these should be enum classes or something.

Reply:
Yes, these are all intended to be exported outside of spirv-opt. They will be used by the Vulkan bindless validation layer to read the data written by the instrumented bindless code and pass it to the Vulkan developer to understand what error occurred and where. Is there a better place for this file?

Reply:
In fact, I had a request to add a few more, which I am going to do next.

Reply:
Is there a bug or design doc I can look at to better understand how this works/ what it does?

Reply:
I have now updated instrument.hpp to make it friendlier to external users/developers. Let me know if you have additional suggestions.